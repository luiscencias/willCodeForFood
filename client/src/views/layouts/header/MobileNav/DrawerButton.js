import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

const StyledIcon = withStyles(theme => ({
	
}))(MenuIcon);

const StyledIconButton = withStyles(theme => ({
	root: {
		color: 'inherit'
	}
}))(IconButton);

const DrawerButton = ({ onClick }) => (
	<StyledIconButton onClick={onClick}>
		<StyledIcon />
	</StyledIconButton>
);

DrawerButton.propTypes = {
	onClick: PropTypes.func.isRequired
};

export default DrawerButton;