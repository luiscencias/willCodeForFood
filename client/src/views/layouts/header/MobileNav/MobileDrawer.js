import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Drawer from '@material-ui/core/Drawer';

const StyledDrawer = withStyles(theme => ({
	paper: {
		minWidth: '60%'
	}
}))(Drawer);

const MobileDrawer = ({ isOpen, onClose, children }) => (
	<StyledDrawer anchor="left" open={isOpen} onClose={onClose}>
		{ children }
	</StyledDrawer>
);

MobileDrawer.propTypes = {
	isOpen: PropTypes.bool.isRequired,
	onClose: PropTypes.func.isRequired,
	children: PropTypes.node.isRequired
};

export default MobileDrawer;