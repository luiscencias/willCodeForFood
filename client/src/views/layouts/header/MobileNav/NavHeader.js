import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import { styledTag } from '../../../../helpers';

import AccountIcon from '@material-ui/icons/AccountCircle';
import Button from '@material-ui/core/Button';

import MainLink from '../MainLink';

const Container = withStyles(theme => ({
	root: {
		display: 'flex',
		flexDirection: 'column',

		paddingBottom: theme.spacing.unitScaleToRem(1),

		backgroundColor: theme.palette.grey[100],

		minHeight: '9.75rem'
	}
}))(styledTag('div'))

const StyledAccountIcon = withStyles(theme => ({
	root: {
		fontSize: '5rem',
		color: theme.palette.grey[300],

		alignSelf: 'center',

		marginTop: theme.spacing.unitScaleToRem(2)
	}
}))(AccountIcon);

const StyledButton = withStyles(theme => ({
	root: {
		marginTop: theme.spacing.unitScaleToRem(2),
		marginBottom: theme.spacing.unitScaleToRem(1),
		marginLeft: theme.spacing.unitScaleToRem(2),
		marginRight: theme.spacing.unitScaleToRem(2),

		minHeight: '2.25rem'
	}
}))(Button);

const NavHeader = ({ handleClick }) => (
	<Container>
		<StyledAccountIcon />
		<StyledButton variant="outlined" component={Link} onClick={handleClick} to="/login">Log In</StyledButton>
	</Container>
);

export default NavHeader;