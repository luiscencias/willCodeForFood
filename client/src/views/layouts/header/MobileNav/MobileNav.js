import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Logo from '../Logo';
import { Spacer, LoginForm } from '../../../../components';
import Login from '../Login';
import ApplyButton from '../ApplyButton';
import NavigationLinks from '../NavigationLinks';

import DrawerButton from './DrawerButton';
import MobileDrawer from './MobileDrawer';

import Divider from '@material-ui/core/Divider';

const MobileNav = ({ isAuthenticated, routes }) => {

	const [ drawerIsOpen, setDrawerIsOpen ] = useState(false);

	return (
		<>
			<MobileDrawer 
				isOpen={ drawerIsOpen }
				onClose={ () => setDrawerIsOpen(false) }
			>
				{ isAuthenticated ? '' : <LoginForm /> }
				<Divider />
				<NavigationLinks handleClick={() => setDrawerIsOpen(false)} routes={[ { name: 'Home', path: '/' }, ...routes]} />
			</MobileDrawer>

			<DrawerButton onClick={ () => setDrawerIsOpen(true) } />

			<Spacer />
			<Logo />
			<Spacer />
			{ isAuthenticated ? '' : <ApplyButton /> }
		</>
	);
};

MobileNav.propTypes = {
	isAuthenticated: PropTypes.bool.isRequired,
	routes: PropTypes.arrayOf(
		PropTypes.shape({ path: PropTypes.string.isRequired, name: PropTypes.string.isRequired })
	).isRequired
};

export default MobileNav;