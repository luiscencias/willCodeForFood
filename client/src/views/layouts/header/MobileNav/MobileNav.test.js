import React from 'react';
import {shallow} from "enzyme/build";
import { Zoom } from 'react-slideshow-image';
import MobileNav from './MobileNav';


describe('MobileNav', () => {

    let mobileNav;
    beforeEach(() => mobileNav = shallow(<MobileNav />));

    it('renders', () => {

        expect(mobileNav.exists()).toBe(true);
    });

    it('renders without changing', () => {
        expect(mobileNav).toMatchSnapshot();
        // console.log(mobileNav.debug());
    });

    it('should render Zoom component', () => {
        expect(mobileNav.contains(<Zoom scale={.4}/>));   //scale is required
        // console.log(mobileNav.debug());
    });

    it('should render an img element', () => {

        expect(mobileNav.contains(<img/>));
    });


    it('should have js implementation', () => {
        // console.log(mobileNav.debug());

        expect(mobileNav.find('.js')).toBeTruthy();
    });
    
});