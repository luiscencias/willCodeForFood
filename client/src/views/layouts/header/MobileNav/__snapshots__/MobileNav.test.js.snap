// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav renders without changing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MobileDrawer
          isOpen={false}
          onClose={[Function]}
        >
          <NavHeader
            handleClick={[Function]}
          />
          <WithStyles(Divider) />
          <NavigationLinks
            handleClick={[Function]}
          />
        </MobileDrawer>,
        <DrawerButton
          onClick={[Function]}
        />,
        <WithStyles(Component) />,
        <Logo />,
        <WithStyles(Component) />,
        <LoginButton />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavHeader
              handleClick={[Function]}
            />,
            <WithStyles(Divider) />,
            <NavigationLinks
              handleClick={[Function]}
            />,
          ],
          "isOpen": false,
          "onClose": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleClick": [Function],
              "routes": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MobileDrawer
            isOpen={false}
            onClose={[Function]}
          >
            <NavHeader
              handleClick={[Function]}
            />
            <WithStyles(Divider) />
            <NavigationLinks
              handleClick={[Function]}
            />
          </MobileDrawer>,
          <DrawerButton
            onClick={[Function]}
          />,
          <WithStyles(Component) />,
          <Logo />,
          <WithStyles(Component) />,
          <LoginButton />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavHeader
                handleClick={[Function]}
              />,
              <WithStyles(Divider) />,
              <NavigationLinks
                handleClick={[Function]}
              />,
            ],
            "isOpen": false,
            "onClose": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleClick": [Function],
                "routes": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
